specification.md

```diff
@@ -117,16 +117,30 @@ These can all be valid timing diagrams for children that "FollowFrom" a parent.
 
 ## The OpenTracing API
 
-There are three critical and inter-related types in the OpenTracing specification: `Tracer`, `Span`, and `SpanContext`. Below, we go through the behaviors of each type; roughly speaking, each behavior becomes a "method" in a typical programming language, though it may actually be a set of related sibling methods due to type overloading and so on.
+There are three critical and inter-related types in the OpenTracing specification: `Tracer`, `Span`, and `SpanContext`, as well as two optional helper types: `ScopeManager` and `Scope`. Below, we go through the behaviors of each type; roughly speaking, each behavior becomes a "method" in a typical programming language, though it may actually be a set of related sibling methods due to type overloading and so on.
 
 When we discuss "optional" parameters, it is understood that different languages have different ways to construe such concepts. For example, in Go we might use the "functional Options" idiom, whereas in Java we might use a builder pattern.
 
+### Within-process `Span` propagation
+
+For any thread or execution unit, at most one `Span` may be active, and it is important that such `Span` is available at any point down the execution chain, in order to define proper **Reference** relationships among `Span`s.
+
+For platforms where the call-context is explicitly propagated down the execution chain -such as `Go`-, such context can be used to store the active `Span` at all times.
+
+For platforms not propagating a call-context, it is inconvenient to pass the active `Span` from function to function manually. For those platforms, the `Tracer` interface must contain a `ScopeManager` instance that stores and handles the active `Span` through a container interface, called `Scope`.
+
 ### `Tracer`
 
-The `Tracer` interface creates `Span`s and understands how to `Inject`
-(serialize) and `Extract` (deserialize) them across process boundaries.
+The `Tracer` interface creates `Span`s, understands how to `Inject`
+(serialize) and `Extract` (deserialize) them across process boundaries, and optionally contains a `ScopeManager` instance to handle the active `Span`.
 Formally, it has the following capabilities:
 
+#### Retrieve the `ScopeManager` (Optional)
+
+There should be no parameters.
+
+**Returns** the used `ScopeManager` instance to set and retrieve the active `Span`. The mentioned active instance is additinaly used by default as the implicit parent for newly created `Span`s, in case no **references** were provided.
```

Comment:
"the active `Span`" -> "the active `Scope`"? or in general I think it should be clear that the `ScopeManager` retrieves a `Scope` that contains the active `Span`. Thoughts?

Reply:
Yeah, I think it makes more sense to make it clear at all moments that `Scope` objects are being returned/used.